plugins/dominant-color-images/hooks.php

```diff
@@ -186,3 +186,76 @@ function dominant_color_render_generator(): void {
+ * to apply background color based on the dominant color for attachment previews
+ * in the WordPress admin interface.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_inline_style(): void {
+	$handle = 'dominant-color-admin-styles';
+	// PHPCS ignore reason: Version not used since this handle is only registered for adding an inline style.
+	// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = '.wp-core-ui .attachment-preview[data-dominant-color]:not(.has-transparency) { background-color: var(--dominant-color); }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_action( 'admin_enqueue_scripts', 'dominant_color_admin_inline_style' );
+
+/**
+ * Adds a script to the admin footer to modify the attachment template.
+ *
+ * This function injects a JavaScript snippet into the admin footer that modifies
+ * the attachment template. It adds attributes for dominant color and transparency
+ * to the template, allowing these properties to be displayed in the media library.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_script(): void {
+	?>
+	<script>
+		(function() {
+			var s = jQuery( '#tmpl-attachment' )
+			var n = s[0].innerText.replace( '{{ data.orientation }}"', '{{ data.orientation }} {{ data.hasTransparencyClass }}"	data-dominant-color="{{ data.dominantColor }}" data-has-transparency="{{ data.hasTransparency }}" style="--dominant-color: #{{ data.dominantColor }};"')
+			s.replaceWith( '<script type="text/html" id="tmpl-attachment">' + n  );
+		}());
+	</script>
+	<?php
+}
+add_action( 'admin_print_footer_scripts', 'dominant_color_admin_script', 1000 );
+
+/**
+ * Prepares attachment data for JavaScript, adding dominant color and transparency information.
+ *
+ * This function enhances the attachment data for JavaScript by including information about
+ * the dominant color and transparency of the image. It modifies the response array to include
+ * these additional properties, which can be used in the media library interface.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<mixed> $response   The current response array for the attachment.
+ * @param WP_Post      $attachment The attachment post object.
+ * @param array<mixed> $meta       The attachment metadata.
+ * @return array<mixed> The modified response array with added dominant color and transparency information.
+ */
+function dominant_color_prepare_attachment_for_js( array $response, WP_Post $attachment, array $meta ): array {
+
+	$response['dominantColor'] = '';
+	if ( isset( $meta['dominant_color'] ) && str_contains( '#', $meta['dominant_color'] ) ) {
+		$response['dominantColor'] = sanitize_hex_color( $meta['dominant_color'] );
+	}
+	$response['hasTransparency']      = '';
+	$response['hasTransparencyClass'] = '';
+	if ( isset( $meta['has_transparency'] ) ) {
+		$response['hasTransparency']      = esc_attr( $meta['has_transparency'] );
```

Comment:
Is `esc_attr()` correct here? As I recall the data here will be passed back as JSON via admin-ajax, and won't this run the risk of accidentally encoding HTML entities? If the the value should be a boolean, would this make more sense as:

```suggestion
		$response['hasTransparency']      = (bool) $meta['has_transparency'];
```

Otherwise, if if it is supposed to be a string then literals could be used as in the line below.